### ISSUE TITLE
Signal folder not visible on Nexus 5x when connecting via USB to Windows computer

### ISSUE THREAD

**Issue by ahelwer** [signalapp/Signal-Android#7760]
- [x ] I have searched open and closed issues for duplicates
- [ x] I am submitting a bug report for existing functionality that does not work as intended
- [x ] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x ] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I am attempting to use the backup/restore workflow specified [here](https://support.signal.org/hc/en-us/articles/360001890291) to transfer my Signal messages from my old Nexus 5x phone to a new phone. Under the "How do I move messages to my new phone?" section, step 2 says to transfer the "Signal" folder from your old phone to your new phone (along with a disclaimer that the process varies by phone/OS). On the Nexus 5x running Android 8.1.0, the Signal folder is not visible from Windows Explorer, even when specifying hidden items should be shown. The signal folder _is_ visible from the Android file system explorer, including all backups contained therein.

Workaround: On your Nexus 5x, go to Settings > Storage > Files and open the Signal directory. Move or copy your Signal backups to the Downloads folder. Connect your Nexus 5x to your computer, and then copy the Signal backup from the Nexus 5x downloads folder to your own computer in preparation for transfer to your new phone. Note that the Signal backup files might show up as 0 KB in size from Windows Explorer; disconnecting then reconnecting my phone to my computer fixed this issue, and the Signal backup files were sized as expected.

### Steps to reproduce
- Create backup from Signal Settings > Chats and Media > Chat backups on your Nexus 5x Android device
- Connect your Nexus 5x to a Windows computer via USB
- On your Nexus 5x, use the USB connection alert to enable file transfer between your computer and device
- Open Windows Explorer and navigate to the root "Internal shared storage" directory; note absence of Signal directory
- On your Nexus 5x, go to Settings > Storage > Files and note the presence of the Signal directory (and backup(s) therein)

**Actual result:** Signal directory is not visible from Windows explorer, hindering attempts to backup/restore Signal messages on another phone.
**Expected result:** Signal directory is visible from Windows explorer, enabling easy copying of Signal directory onto your computer and then onward to your new phone.

### Screenshots

![Windows Explorer view (note Hidden items is checked)](https://user-images.githubusercontent.com/718765/39603858-bf6a7c7e-4ede-11e8-8e45-840cee550af9.png)

![Android file explorer view](https://user-images.githubusercontent.com/718765/39603943-03f37dc8-4edf-11e8-88f5-7f0eec660678.png)

### Device info
**Device:** LG Nexus 5x 32 GB
**Android version:** 8.1.0
**Signal version:** 4.18.3

### Link to debug log
https://debuglogs.org/9ac618bbe6bdf40c02e730a3d6c4c8f25b377bc2a109af36a548d797fc32c2ba

**Comment by ahelwer** [comment]
Secondary minor bug (or gap in the restore instructions) - on my Nexus 5x, the directory structure was /Signal/*.backup but in order to be auto-detected as a backup on the new phone the directory structure must be /Signal/backups/*.backup.

**Comment by igor2x** [comment]
I agree it is big annoyance you can't quickly copy/paste data to new device, but this issue in my humble opinion does not satisfy the [new rules](https://github.com/signalapp/Signal-Android/issues/7598) condition for opening bug report: _If you disagree with the way that a feature is working that's not a bug._

@ahelwer, I also have Nexus 5x and from Setting | Storage | Files I can see Signal/Backup directory and backups stored in there.

By the way, if data can be seen with Android explorer and can't be with Windows file manager, this may be an issue in Windows...

May I kindly suggest this issue to be closed down as bug report and invite you to continue discussion on [community forum](https://community.signalusers.org/).

**Comment by ahelwer** [comment]
The bug here is that I was attempting to follow the backup/restore instructions and ran into two separate pitfalls which likely would have frustrated a less-technical user and kept the feature from working. I don't "disagree" with the way the feature works, I am reporting that the feature does not work as documented. Either the documentation is wrong or the feature has a bug. In complicated features like a backup-restore workflow involving, at minimum, two separate mobile devices (possibly with two separate mobile OSs) and a computer running a fully different OS, these tiny rough patches are absolutely bugs that must be fixed in order for the feature to work.

**Comment by piratenpanda** [comment]
Same for me on Arch Linux so definitely not a problem with Windows Explorer. I need to boot to TWRP to restore the backup so I'd also like to see this fixed for not so technically inclined people.

@igor2x would be nice if you'd actually read and understand the bug report and not act like some kind of authority which you are not. The bug is annoying for everyone and not only Windows users.

**Comment by Hormesis** [comment]
I've had the same issue across multiple phones. An invisible folder is very much a bug, not a feature.

**Comment by siccovansas** [comment]
I think this might not be a Signal issue, but rather the following Android issue: [Android MTP support does not show recent files until the device is rebooted](https://issuetracker.google.com/u/1/issues/36956498).

My fix is to run SD Scanner ([F-Droid](https://f-droid.org/en/packages/com.gmail.jerickson314.sdscanner/), [Google Play](https://play.google.com/store/apps/details?id=com.gmail.jerickson314.sdscanner&hl=en)) after I created an export/backup and the folders/files will show up on your computer.

**Comment by ahelwer** [comment]
Neat, thanks @siccovansas ! I'll try testing this theory on my phone when I get a chance. I don't have my Nexus 5x anymore, but maybe I can repro the invisible folder issue on my new phone then see if rebooting fixes it.

**Comment by piratenpanda** [comment]
Seems indeed to be the problem @siccovansas

**Comment by 2-4601** [comment]
Thanks for confirming. I haven't tried the new backup feature in Signal but I've run into this Android bug @siccovansas referenced to many times.

**Comment by haffenloher** [comment]
As this "SD Scanner" app mentioned by @siccovansas doesn't seem to require root, maybe there's a way for us to do the same thing this app does to work around the Android bug? If not, this problem and the non-trivial workaround (rebooting your phone) should at least be mentioned in a support article.

**Comment by 2-4601** [comment]
https://github.com/jerickson314/sdscanner

**Comment by StrawberryPearl** [comment]
@ahelwer I need to tell you, your post saved my sanity. I was switching from a Droid Moto to a Pixel 3a and I nearly lost my mind trying to figure out how to get the Signal backup file from the old phone to the new phone. It is absolutely a bug in Signal - there is no way it needed to be as difficult and time consuming as it was. Thank you, thank you, thank you, a million times, thank you.
